src/main/java/com/epam/ta/reportportal/ws/controller/UserController.java

```diff
@@ -2,32 +2,198 @@
+			@AuthenticationPrincipal ReportPortalUser currentUser, HttpServletRequest request) {
+		/*
+		 * Use Uri components since they are aware of x-forwarded-host headers
+		 */
+		URI rqUrl = UriComponentsBuilder.fromHttpRequest(new ServletServerHttpRequest(request))
+				.replacePath(null)
+				.replaceQuery(null)
+				.build()
+				.toUri();
+		return createUserMessageHandler.createUserBid(createUserRQ, currentUser, rqUrl.toASCIIString());
+	}
+
+	@PostMapping(value = "/registration")
+	@ResponseStatus(CREATED)
+	@ApiOperation("Activate invitation and create user in system")
+	public CreateUserRS createUser(@RequestBody @Validated CreateUserRQConfirm request, @RequestParam(value = "uuid") String uuid,
+			@AuthenticationPrincipal ReportPortalUser currentUser) {
+		return createUserMessageHandler.createUser(request, uuid, currentUser);
+	}
+
+	@GetMapping(value = "/registration")
+	@ApiIgnore
+	public UserBidRS getUserBidInfo(@RequestParam(value = "uuid") String uuid) {
+		return getUserHandler.getBidInformation(uuid);
+	}
+
+	@DeleteMapping(value = "/{login}")
+	//@PreAuthorize(ADMIN_ONLY)
+	@ApiOperation(value = "Delete specified user", notes = "Allowable only for users with administrator role")
+	public OperationCompletionRS deleteUser(@PathVariable String login, @AuthenticationPrincipal ReportPortalUser currentUser) {
+		return deleteUserMessageHandler.deleteUser(EntityUtils.normalizeId(login), currentUser);
+	}
+
+	@PutMapping(value = "/{login}")
+	//@PreAuthorize(ALLOWED_TO_EDIT_USER)
+	@ApiOperation(value = "Edit specified user", notes = "Only for administrators and profile's owner")
+	public OperationCompletionRS editUser(@PathVariable String login, @RequestBody @Validated EditUserRQ editUserRQ,
+			@ActiveRole UserRole role, @AuthenticationPrincipal ReportPortalUser currentUser) {
+		return editUserMessageHandler.editUser(EntityUtils.normalizeId(login), editUserRQ, role);
+	}
+
+	@GetMapping(value = "/{login}")
+	@ResponseView(ModelViews.FullUserView.class)
+	//@PreAuthorize(ALLOWED_TO_EDIT_USER)
+	@ApiOperation(value = "Return information about specified user", notes = "Only for administrators and profile's owner")
+	public UserResource getUser(@PathVariable String login, @AuthenticationPrincipal ReportPortalUser currentUser) {
+		return getUserHandler.getUser(EntityUtils.normalizeId(login), currentUser);
+	}
 
 	@Transactional(readOnly = true)
 	@GetMapping(value = { "", "/" })
 	@ApiOperation("Return information about current logged-in user")
-	public UserResource getMyself(@AuthenticationPrincipal ReportPortalUser user) {
-		return UserConverter.TO_RESOURCE.apply(userRepository.findByLogin(EntityUtils.normalizeId(user.getUsername()))
-				.orElseThrow(() -> new ReportPortalException(ErrorType.USER_NOT_FOUND, user.getUsername())));
+	public UserResource getMyself(@AuthenticationPrincipal ReportPortalUser currentUser) {
+		return getUserHandler.getUser(currentUser);
+	}
+
+	@GetMapping(value = "/all")
```

Comment:
you need to have in such methods @Transactional(readOnly = true) 
investigate this moment deeply

Reply:
have you checked all your get methods ?